docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,276 @@
+```
+</details>
+
+
+:::note
+Our code logic relies on obtaining connection information from the environment variables in the system and `--binding-as-files=false` ensures that. Read [here](../../command-reference/add-binding.md#understanding-bind-as-files) to know more about this flag.
+:::
+
+Wait for `odo dev` to detect the new changes to `devfile.yaml`. 
+
+<details>
+<summary>Sample output:</summary>
+
+```shell
+$ odo dev
+  __
+ /  \__     Developing using the my-go-app Devfile
+ \__/  \    Namespace: odo-dev
+ /  \__/    odo version: v3.0.0-rc1
+ \__/
+
+↪ Deploying to the cluster in developer mode
+•  Waiting for Kubernetes resources  ...
+⚠  Pod is Pending
+✓  Pod is Running
+✓  Syncing files into the container [152ms]
+✓  Building your application in container on cluster (command: build) [15s]
+•  Executing the application (command: run)  ...
+-  Forwarding from 127.0.0.1:40001 -> 8080
+
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
+File /tmp/go/.odo changed
+•  Waiting for Kubernetes resources  ...
+✓  Syncing files into the container [1ms]
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
+File /tmp/go/devfile.yaml changed
+•  Waiting for Kubernetes resources  ...
+✓  Creating kind ServiceBinding
+Error occurred on Push - watch command was unable to push component: some servicebindings are not injected
+
+Updating Component...
+
+•  Waiting for Kubernetes resources  ...
+Error occurred on Push - watch command was unable to push component: some servicebindings are not injected
+
+⚠  Pod is Terminating
+•  Waiting for Kubernetes resources  ...
+✗  Finished executing the application (command: run) [1m]
+⚠  No pod exists
+⚠  Pod is Pending
+✓  Pod is Running
+✓  Syncing files into the container [170ms]
+✓  Building your application in container on cluster (command: build) [192ms]
+•  Executing the application (command: run)  ...
+-  Forwarding from 127.0.0.1:40001 -> 8080
+
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+Pushing files...
+
+
+File /tmp/go/devfile.yaml changed
+
+File /tmp/go/.odo/devstate.json changed
+•  Waiting for Kubernetes resources  ...
+✓  Syncing files into the container [1ms]
+
+Watching for changes in the current directory /tmp/go
+Press Ctrl+c to exit `odo dev` and delete resources from the cluster
+
+
+```
+</details>
+
+
+## Step 4. Check the connection again
+Query the URL again for a successful connection: 
+```shell
+curl 127.0.0.1:40001
+```
+
+<details>
+<summary>Sample output:</summary>
+
+```shell
+$ curl 127.0.0.1:40001
+Successfully connected and pinged.
+```
+</details>
+
+
+## Step 5. Exit and cleanup
+Press `Ctrl+c` to exit `odo dev`.
+
+Delete the MongoDB instance that we had created.
```

Comment:
Stupid question, but why are we using kubectl / oc commands for deleting this mongodb instance?

Because `odo delete binding` isn't added yet right?

We should add a note that we intent on implementing this so we update our documentation accordingly in the future.

Reply:
I do not think we intend on implementing `odo delete binding`. We already have `odo remove binding` that will remove the binding from devfile.yaml, and eventually it will delete the ServiceBinding instance from cluster, but the mongodb instance stays the same.
Since we created the mongodb instance at the beginning, it's only fair that we remove it.